docs/api.md

```diff
@@ -143,3 +152,133 @@ Only app owners or co-maintainers are allowed to delete an app release. The owne
   * **HTTP 401**: If the user is not authenticated
   * **HTTP 403**: If the user is not authorized to delete the app release
 
+
+## Release a New App Release Or Update an Existing One
+The following request will create a new app release:
+
+* **Url**: PUT /api/v1/apps/{app-id}/releases/{version}
+  * **Parameters**:
+    * **app-id**: lowercase ASCII characters and underscores, the app's id
+    * **version**: semantic version, digits only: app release version
+* **Content-Type**: application/json
+* **Request body**:
+  * **download**: An Https (Http is not allowed!) link to the archive packaged (maximum size: 50 Megabytes) as tar.gz:
+
+  ```json
+  {"download":"https://example.com/release.tar.gz"}
+  ```
+
+* **Example CURL request**:
+
+        curl -X PUT -u "user:password" http://localhost:8000/api/v1/apps/news/releases -H "Content-Type: application/json" -d '{"download":"https://example.com/release.tar.gz"}'
+
+* **Returns**:
+  * **HTTP 200**: If the app release was update successfully
+  * **HTTP 201**: If the app release was created successfully
+  * **HTTP 400**: If the app contains invalid data, is too large or could not be downloaded
+  * **HTTP 401**: If the user is not authenticated
+  * **HTTP 403**: If the user is not authorized to delete the app release
+
+If there is not yet an app with the given app id yet, a new app is created and the owner is set in to the logged in user. Then the **info.xml** file which lies in the compressed archive's folder **app-id/appinfo/info.xml** is being parsed and validated.
+
+Afterwards the app and app release will be created or updated, depending on if they already exist. Both owners and co-maintainers are allowed to upload new releases.
```

Comment:
This is different to what you write above. Above you wirte "if not exist create app and download archiv and read info.xml. Afterwards app and release will be created. 
So what actually happen download and then create/update or create and then download?